src/hooks/common/inputsUserInfo/useCheckNicknameChanged.ts

```diff
@@ -0,0 +1,58 @@
+import { useRef, useState } from 'react';
+import { AxiosErrorResponse } from '@/api/interceptor';
+import { getNicknameVerify } from '@/api/member';
+import { HTTP_STATUS_CODE } from '@/constants/api';
+import { ERROR, SUCCESS } from '@/constants/message';
+import { isEmptyString } from '@/utils/validator';
+import { useMutation } from '@tanstack/react-query';
+
+export const useCheckNicknameChanged = (
+  value: string,
+  defaultValue: string,
+  setIsNicknameSuccess: React.Dispatch<React.SetStateAction<boolean>>,
+) => {
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const [errorMessage, setErrorMessage] = useState<string | undefined>(
+    undefined,
+  );
+  const [isError, setIsError] = useState<boolean>(false);
+
+  function isValidNickname(nickname: string): boolean {
+    return nickname.length >= 2 && nickname.length <= 10;
+  }
+
+  const nicknameVerify = useMutation({
+    mutationFn: () => getNicknameVerify(value),
+    onSuccess: () => {
+      setIsError(false);
+      setErrorMessage(SUCCESS.NICKNAME.AVAILABLE);
+      setIsNicknameSuccess(true);
+    },
+    onError: (err: AxiosErrorResponse) => {
+      if (err.status === HTTP_STATUS_CODE.CONFLICT) {
+        setIsError(true);
+        setErrorMessage(ERROR.NICKNAME.EXIST);
+        setIsNicknameSuccess(false);
+      }
+    },
+  });
+
+  const handleSubmit = () => {
+    if (isEmptyString(value)) {
+      return;
+    }
+
+    if (value === defaultValue) {
+      return;
+    }
+
+    if (!isValidNickname(value)) {
+      setIsError(true);
+      setErrorMessage(ERROR.NICKNAME.FORM);
+    } else {
+      nicknameVerify.mutate();
+    }
+  };

```

Comment:
```suggestion
    if (!isValidNickname(value)) {
      setIsError(true);
      setErrorMessage(ERROR.NICKNAME.FORM);
      return;
}

      nicknameVerify.mutate();
  };
```
가독성을 위해 이런 식의 early return 을 사용해도 좋아보여요~

Reply:
[9839298](https://github.com/JECT-Study/PICK-O-Client/pull/244/commits/9839298247398530318a53ac6cbb609d4d84d0b9) 수정했습니다!!